find_match.py

```diff
@@ -1,10 +1,10 @@
-def find_match(model, product_a_id):
-    max_score_and_product_id = (0, None)
+def find_match(model, product_a_id, range):
+	top_matches = []
 
-    for product_b in model.products:
-        new_score = model.score(product_a_id, product_b)
-
-        if new_score > max_score_and_product_id[0]:
-            max_score_and_product_id = (new_score, product_b)
-
-    return max_score_and_product_id
+	for product_b in model.products:
+		new_score = model.score(product_a_id, product_b)
+		if new_score > top_matches[-1][1]
+			top_matches.append((new_score, product_b))
+			top_matches.sort(key=lambda tup: -tup[0])
```

Comment:
Du behöver egentligen bara sortera listan som det sista du gör och spara undan vilken som är sämst i din lista just nu. Om det är mindre än `range` i listan så lägger du alltid in.